src/Microsoft.IdentityModel.Protocols.SignedHttpRequest/Exceptions/SignedHttpRequestCreationException.cs

```diff
@@ -0,0 +1,63 @@
+﻿//------------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+using System;
+
+namespace Microsoft.IdentityModel.Protocols.SignedHttpRequest
```

Comment:
In general for desktop for exceptions to support serialization we need to have the following
```
#if DESKTOPNET45  
	[Serializable]  
#endif
```
Then add method

```
#if DESKTOPNET45
        /// <summary>
        /// Initializes a new instance of the <see cref="SignedHttpRequestCreationException"/> class.
        /// </summary>
        /// <param name="info">the <see cref="SerializationInfo"/> that holds the serialized object data.</param>
        /// <param name="context">The contextual information about the source or destination.</param>
        protected SignedHttpRequestCreationException(SerializationInfo info, StreamingContext context)
            : base(info, context)
        {
        }
#endif
```

Reply:
I didn't put this code in as I thought that is never used. 
I'm not sure if the DESKTOPNET45 is a correct target i.e. will this code ever be compiled?

Reply:
OH, we should make sure that this code is in .NET DESKTOP.

Reply:
I'm not sure if this code ever worked. Even if there was a DESKTOPNET45 directive defined, I think that this code wouldn't compile.

Reply:
todo: open up a new issue on custom exception serialization.